From 6d053ec0a4108be59f5e30a04ab4e6539e694e53 Mon Sep 17 00:00:00 2001
From: Simon McVittie <smcv@debian.org>
Date: Fri, 18 Sep 2020 11:31:52 +0100
Subject: [PATCH] tests: Explicitly depend on the GIR and typelib

In Meson since 0.55.0, having the test executable depend on the GIR
dependency is not enough to ensure that uninstalled GIR and typelib
files are generated: we also need to have the test itself depend on
the GIR build target.

It is not clear to me whether this is a workaround for a Meson
regression (if what libpeas previously did was meant to work), or a
solution to a libpeas bug (if it was never meant to be guaranteed to
work). For more details see
<https://github.com/mesonbuild/meson/issues/7756>.

Resolves: https://gitlab.gnome.org/GNOME/libpeas/-/issues/40
Bug-Debian: https://bugs.debian.org/966951
---
 tests/libpeas/meson.build | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/tests/libpeas/meson.build b/tests/libpeas/meson.build
index 32cf5e9..c95190a 100644
--- a/tests/libpeas/meson.build
+++ b/tests/libpeas/meson.build
@@ -53,5 +53,10 @@ foreach test: libpeas_tests_sources
     c_args: libpeas_tests_c_args,
   )
 
-  test('test-@0@'.format(test_name), test_exe, env: test_env)
+  test(
+    'test-@0@'.format(test_name),
+    test_exe,
+    depends: [libintrospection_gir],
+    env: test_env,
+  )
 endforeach
-- 
2.26.2

